pennylane_qulacs/qulacs_device.py

```diff
@@ -192,6 +173,23 @@ def apply_operations(self, operations):
             else:
                 self._apply_gate(op)
 
+    def _expand_state(self, state_vector, wires):
+        """Expands state vector to more wires"""
+        basis_states = np.array(list(it.product([0, 1], repeat=len(wires))))
+
+        # get basis states to alter on full set of qubits
+        unravelled_indices = np.zeros((2 ** len(wires), self.num_wires), dtype=int)
+        unravelled_indices[:, wires] = basis_states
+
+        # get indices for which the state is changed to input state vector elements
+        ravelled_indices = np.ravel_multi_index(unravelled_indices.T, [2] * self.num_wires)
+
+        state_vector = self._scatter(ravelled_indices, state_vector, [2 ** self.num_wires])
+        state_vector = self._reshape(state_vector, [2] * self.num_wires)
+        state_vector = self._asarray(state_vector, dtype=self.C_DTYPE)
```

Comment:
The method looks nice, and easy to follow!

This is relatively minor, but you don't necessarily need to constrain yourself to these staticmethods for array manipulation, feel free to use standard NumPy 🙂 

The main purposes of these static methods was to support different array manipulation libraries without needing to re-write all the logic. However,

- The qulacs devices only use NumPy anyway, so it is fine to use numpy directly here
- We are likely going to remove these static methods soon anyway, in favour of `qml.math`.

So I recommend replacing these with `np` to avoid needing to change it later down the line.

Reply:
Good point. Copy-pasted this part of the code from default.qubit, so that's why it stuck. Make sense removing it in favour of NumPy. 👍